BraintreeCore/src/main/java/com/braintreepayments/api/AuthorizationLoader.java

```diff
@@ -0,0 +1,59 @@
+package com.braintreepayments.api;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+class AuthorizationLoader {
+
+    private Authorization authorization;
+    private final ClientTokenProvider clientTokenProvider;
+
+    AuthorizationLoader(@Nullable String initialAuthString, @Nullable ClientTokenProvider clientTokenProvider) {
+        this.clientTokenProvider = clientTokenProvider;
+        if (initialAuthString != null) {
+            this.authorization = Authorization.fromString(initialAuthString);
+        }
+    }
+
+    void loadAuthorization(@NonNull final AuthorizationCallback callback) {
+        if (authorization != null) {
+            callback.onAuthorizationResult(authorization, null);
```

Comment:
Do we need to do anything about client token expiration or is that up to the merchant? What happens here if we have an authorization but it's no longer valid?

Reply:
I believe in `v3` and `v4` we assume the merchant has recently fetched a client token and we keep a reference for as long as the SDK client / BraintreeFragment is in memory. This mimics that behavior since the auth provider is owned by the `BraintreeClient` instance, they'll be de-allocated at the same time.

Reply:
Expired tokens will result in an error callback and that should be at parity with how it works now. AFAIK we haven't considered caching client tokens but this opens up the possibility. We can gather more information about client token caching we can make this a feature in another minor version.